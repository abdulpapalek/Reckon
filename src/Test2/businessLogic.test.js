/* eslint-disable */
const BusinessLogic = require("./businessLogic");
const Api = require("./api");

describe("BusinessLogic", () => {  
    afterEach(() => {
      // cleaning up the mess left behind the previous test
      jest.clearAllMocks();
    });
  
    test("Success: Should be able to send submit correctly", async () => {
        Api.getSubtexts = jest.fn(async () => Promise.resolve({
            subTexts: [
              "Peter",
              "peter",
              "Pick",
              "Pi",
              "Z"
            ]
          }
        ));
        Api.getTextToSearch = jest.fn(async () => Promise.resolve({
          text: "Peter told me (actually he slurrred) that peter the pickle piper piped a pitted pickle before he petered out. Phew!"
          }
        ));
        expect(await BusinessLogic.persistResults()).toMatchSnapshot();;
    });
});