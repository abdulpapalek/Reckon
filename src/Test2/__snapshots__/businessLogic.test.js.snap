// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessLogic Success: Should be able to send submit correctly 1`] = `
Object {
  "candidate": "Alexander Kevin R. Amante",
  "results": Array [
    Object {
      "result": "1, 43, 98",
      "subtext": "Peter",
    },
    Object {
      "result": "1, 43, 98",
      "subtext": "peter",
    },
    Object {
      "result": "53, 81",
      "subtext": "Pick",
    },
    Object {
      "result": "53, 60, 66, 74, 81",
      "subtext": "Pi",
    },
    Object {
      "result": "<No Output>",
      "subtext": "Z",
    },
  ],
  "text": "Peter told me (actually he slurrred) that peter the pickle piper piped a pitted pickle before he petered out. Phew!",
}
`;
